apps/pollbook/frontend/src/app_poll_worker_screen.test.tsx

```diff
@@ -142,100 +142,78 @@ describe('PollWorkerScreen', () => {
     unmount();
   });
 
-  test.skip('primary election e2e flow with party selection', async () => {
+  test('primary election happy path e2e flow with party selection', async () => {
     apiMock.expectGetDeviceStatuses();
-    apiMock.authenticateAsPollWorker(multiPartyPrimaryElection);
-    apiMock.setElection(multiPartyPrimaryElectionDefinition);
-    const { unmount } = render(<App apiClient={apiMock.mockApiClient} />);
-    await screen.findByText('Connect printer to continue.');
-
     apiMock.setPrinterStatus(true);
     apiMock.setIsAbsenteeMode(false);
-    await screen.findByText('No Precinct Selected');
-
+    apiMock.authenticateAsPollWorker(multiPartyPrimaryElection);
     apiMock.setElection(
       multiPartyPrimaryElectionDefinition,
       multiPartyPrimaryElection.precincts[0].id
     );
-
-    apiMock.expectGetCheckInCounts({ allMachines: 25, thisMachine: 5 });
-    await vi.waitFor(() => {
-      screen.getByText('Check-In');
-    });
     apiMock.expectGetScannedIdDocument();
     apiMock.expectSearchVotersNull({});
```

Comment:
I believe the main issue was this mock being set up after the actual render. The rest of the changes to this test are just deleting checks that are performed in the general election test above.